torch_geometric/explain/algorithm/pgm_explainer.py

```diff
@@ -0,0 +1,440 @@
+
+        # build explanation
+        return Explanation(
+            x=x,
+            edge_index=edge_index,
+            markov_blanket=neighbors,
+            # pgm_explation=pgm_explanation,
+            # marginal_probability=marginal_prob,
+            pgm_stats=pgm_stats,
+        )
+
+    def batch_perturb_features_on_node(
+            self,
+            num_samples,
+            index_to_perturb,
+            X_features,
+            model,
+            percentage=50,  # % time node gets pertubed
+            p_threshold=0.05,
+            pred_threshold=0.1,
+            pertubation_mode="mean",
+            snorm_node=None,
+            snorm_edge=None,
+            **kwargs):
+        # for pertubing a batch of graphs for graph classification tasks
+
+        pred_torch = model(X_features, **kwargs)
+        soft_pred = torch.softmax(pred_torch, dim=1)
+        pred_label = torch.argmax(soft_pred, dim=1)
+        num_nodes = X_features.shape[0]
+
+        samples = []
+        for iteration in range(num_samples):
+            X_perturb = X_features.detach().clone()
+            sample = []
+            for node in range(num_nodes):
+                if node in index_to_perturb:
+                    seed = np.random.randint(100)
+                    if seed < percentage:
+                        latent = 1
+                        X_perturb = self.perturb_features_on_node(
+                            X_perturb, node, pertubation_mode=pertubation_mode,
+                            perturb_feature_list=self.perturb_feature_list,
+                            is_random=True)
+                    else:
+                        latent = 0
+                else:
+                    latent = 0
+                sample.append(latent)
+            # if data:
+            #     data.x = X_perturb.detach().clone()
+            # kwargs['data'] = data
+            pred_perturb_torch = model(X_perturb, **kwargs)
+            soft_pred_perturb = torch.softmax(pred_perturb_torch,
+                                              dim=1).squeeze()
+
+            pred_change = torch.max(soft_pred) - soft_pred_perturb[pred_label]
+
+            sample.append(pred_change)
+            samples.append(sample)
+
+        samples = torch.tensor(samples)
+        if self.perturb_indicator == "abs":
+            samples = torch.abs(samples)
+
+        top = int(num_samples / 8)
+        top_idx = torch.argsort(samples[:, num_nodes])[-top:]
+        for i in range(num_samples):
+            if i in top_idx:
+                samples[i, num_nodes] = 1
+            else:
+                samples[i, num_nodes] = 0
+
+        return samples
+
+    def _explain_graph(
+        self,
+        model,
+        x,
+        index,
+        target=None,
+        num_samples=100,
+        top_node=None,  # num of neightbours to consider
+        significance_threshold=0.05,
+        pred_threshold=0.1,
+        percentage=0.5,
+        pertubation_mode="mean",
```

Comment:
Better to specify the types of all arguments, like what you did in `_explain_node()`